Formula/f/fileql.rb

```diff
@@ -0,0 +1,18 @@
+class Fileql < Formula
+  desc "Run SQL-like query on local files instead of database files using the GitQL SDK"
+  homepage "https://github.com/AmrDeveloper/FileQL"
+  url "https://github.com/AmrDeveloper/FileQL/archive/refs/tags/0.6.0.tar.gz"
+  sha256 "cf0a420cf0cd634fe9b60e63718373458eeb25f210e1d2886791aa8a8703a2d7"
+  license "MIT"
+  head "https://github.com/AmrDeveloper/FileQL.git", branch: "master"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    system bin/"fileql", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate temasterath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.